misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/mock/FunctionMock.java

```diff
@@ -34,6 +35,11 @@ public static BError thenReturn(BObject caseObj) {
         BObject mockFunctionObj = caseObj.getObjectValue(StringUtils.fromString("mockFuncObj"));
         BArray args = caseObj.getArrayValue(StringUtils.fromString("args"));
         Object returnVal = caseObj.get(StringUtils.fromString("returnValue"));
+        if (returnVal.toString().equals(MockConstants.FUNCTION_CALLORIGINAL_PLACEHOLDER)) {
+            TesterinaUtils.isCallOriginal = true;
+        } else {
+            TesterinaUtils.isCallOriginal = false;
+        }
```

Comment:
This will break if we use calllOriginal for one mock function and another with diffrent function. We need to use function name vs callOriginal state map

Reply:
Created a separate task https://github.com/ballerina-platform/ballerina-lang/issues/35764 to address this